lib/Basic/Subprocess.cpp

```diff
@@ -421,23 +409,35 @@ void llbuild::basic::spawnProcess(
 
   // Form the complete C string command line.
   std::vector<std::string> argsStorage(commandLine.begin(), commandLine.end());
+#if defined(_WIN32)
+  std::string args;
+  std::for_each(std::begin(argsStorage) + 1, std::end(argsStorage),
+                [&args](auto arg) { args += arg + " "; });
```

Comment:
This is a really hard problem because the command line formatting rules on Windows are quite complex. You essentially need to do the inverse of `CommandLineToArgvW` here, which isn't something provided by the Windows API. Does LLVM have an implementation of converting a string array to a string suitable for passing to `CreateProcess`?

There's also a great MSDN article which details the process for how to do this: https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/

Reply:
LLVM exposes a handy `std::string flattenWindowsCommandLine(ArrayRef<StringRef> Args);`, but only in a more [recent version (mid 2018)](https://github.com/llvm-mirror/llvm/commit/07d33bf7d7df1df00c40bdc859c17025d8457cf8) than what's in [llbuild](https://github.com/apple/swift-llbuild/commit/089f9922c6cdf68cee068593ba805fb4dceb55ea). What's the process like for updating the included llvm support libraries?

Reply:
@ddunbar, can you advise?

Reply:
@ddunbar We should copy the code from the apple/swift-llvm repo, right?

Reply:
Yes. See `lib/llvm/import-llvm` for a little more info

Reply:
I would suggest decoupling that from this particular PR though, we can just start with a not 100% functional version and improve it when the LLVM code lands